### ISSUE TITLE
Remove parse-has-disallow-nesting-has-inside-has.html forgiving selector tests

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/interop#598]
### Test List

https://wpt.fyi/results/css/selectors/parsing/parse-has-disallow-nesting-has-inside-has.html, either the whole test or the subtests ':has(:is(:has(*)))' and ':has(:where(:has(*)))'.

### Rationale

The tests were added in https://github.com/web-platform-tests/wpt/pull/42314 following a resolution in https://github.com/w3c/csswg-drafts/issues/8356. The spec change hasn't been made yet.

This is a scope change and even with a spec change we can't commit to fixing this in Chrome before EOY.

If we carry `:has()` over into Interop 2024, then adding the test back there is probably OK. (I'd have to confirm with the team.)

**Comment by foolip** [comment]
The last subtest in [parse-has.html](https://wpt.fyi/results/css/selectors/parsing/parse-has.html) ('":has(:is(.a, 123))" should be a valid selector') should also be removed for the same reason.

**Comment by foolip** [comment]
It's been a while, but summarizing the discussions from https://github.com/web-platform-tests/interop/issues/599.

The distilled argument I'm making is that the tests were changed, and if we had reviewed the change ahead of time, we would not have had consensus to expand the scope of Interop 2023 with those new tests, and suggested that the tests be added in some other files.

And the reason we're making the argument is that @lilles's team had already finished the work on this focus area for Interop 2023, so the regression was from 100% to less than 100%.

I probed for objections to us sending a PR to split out these tests, and nobody objected.

I hope and believe that's a fair representation of what was discussed.
